src/result-async.ts

```diff
@@ -191,6 +191,16 @@ export class ResultAsync<T, E> implements PromiseLike<Result<T, E>> {
     return this._promise.then((res) => res.unwrapOr(t))
   }
 
+  unwrapOrElse<A>(err: (e: E) => A): Promise<T | A> {
+    return this._promise.then((res) => {
+      if (res.isErr()) {
+        return err(res.error)
```

Comment:
reading this was a little confusing because of the `err` function in the library :)

Reply:
Indeed. Rereading, I was confused too 😆 

I made it better 😉 

https://github.com/supermacro/neverthrow/pull/593/commits/0f34663e2f2310e41affcb56d14b1bd9ac823a5a